### PR TITLE
[FEATURE] Add module type

### PR DESCRIPTION
**Pull request by RandomByte**
Prerequisite for third-party shimming based on https://github.com/SAP/ui5-tooling/pull/5

Depends on https://github.com/SAP/ui5-project/pull/46

Example implementation: https://github.com/RandomByte/openui5-parallax/pull/1/files#diff-f7cafeb3d075b6c5d28c1102cec23f52R14

### CODE REVIEW

File: lib/types/module/ModuleBuilder.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class ModuleBuilder extends AbstractBuilder {
+	constructor({resourceCollections, project, parentLogger}) {
+		super({project, parentLogger});
+
+		// All available library tasks in execution order
+		this.availableTasks = [
+			"uglify"
+		];
+
+		this.addTask("uglify", () => {
```
Comment (matz3) [inline lib/types/module/ModuleBuilder.js]: Do we really want to always uglify?
I would expect that modules from the npm registry are already optimized.

Comment (RandomByte) [reply lib/types/module/ModuleBuilder.js]: Done

**Comment by RandomByte** [comment]
I think we should tackle increasing the test coverage of the types separately. While the "module" type has a coverage of ~20% now, the others have similarly bad ~50%. There are no dedicated tests for the types yet.

**Comment by RandomByte** [comment]
@matz3 ping

**Pull request merged**